components/jwt-sidecar/get-vault-secrets.sh

```diff
@@ -148,44 +157,59 @@ run()
   NS=$(echo $RESP | jq -r '.auth.metadata.namespace')
 
   echo "Getting $SECNAME from Vault $VAULT_PATH and output to $LOCPATH"
-  if [ "$VAULT_PATH" == "secret/tsi-rcni" ]; then
-    CMD="vault kv get -format=json ${VAULT_PATH}/${REGION}/${CLUSTER}/${NS}/${IMGSHA}/${SECNAME}"
-  elif [ "$VAULT_PATH" == "secret/tsi-r" ]; then
-    CMD="vault kv get -format=json ${VAULT_PATH}/${REGION}/${SECNAME}"
-  elif [ "$VAULT_PATH" == "secret/tsi-ri" ]; then
-    CMD="vault kv get -format=json ${VAULT_PATH}/${REGION}/${IMGSHA}/${SECNAME}"
-  elif [ "$VAULT_PATH" == "secret/tsi-rcn" ]; then
-    CMD="vault kv get -format=json ${VAULT_PATH}/${REGION}/${CLUSTER}/${NS}/${SECNAME}"
+  if [ "$VAULT_PATH" == "tsi-rcni" ]; then
+    # CMD="vault kv get -format=json ${VAULT_PATH}/${REGION}/${CLUSTER}/${NS}/${IMGSHA}/${SECNAME}"
+    CMD="${VAULT_ADDR}/v1/secret/data/${VAULT_PATH}/${REGION}/${CLUSTER}/${NS}/${IMGSHA}/${SECNAME}"
+  elif [ "$VAULT_PATH" == "tsi-r" ]; then
+    CMD="${VAULT_ADDR}/v1/secret/data/${VAULT_PATH}/${REGION}/${SECNAME}"
+  elif [ "$VAULT_PATH" == "tsi-ri" ]; then
+    CMD="${VAULT_ADDR}/v1/secret/data/${VAULT_PATH}/${REGION}/${IMGSHA}/${SECNAME}"
+  elif [ "$VAULT_PATH" == "tsi-rcn" ]; then
+    CMD="${VAULT_ADDR}/v1/secret/data/${VAULT_PATH}/${REGION}/${CLUSTER}/${NS}/${SECNAME}"
   else
     echo "Unknown Vault path value!"
-    rm -rf "${SECOUTDIR}/${LOCPATH}/${SECNAME}"
+    rm -rf "${SECFILE}"
     return 1
   fi
 
-  #CMD="vault kv get -format=json secret/tsi-rcni/${REGION}/${CLUSTER}/${NS}/${IMGSHA}/${SECNAME}"
-  #vault kv get -format=json secret/tsi-rcni/eu-de/ti-test/trusted-identity/f36b6d491e0a62cb704aea74d65fabf1f7130832e9f32d0771de1d7c727a79cc/dummy | jq -c '.data.data'
+  # CMD="vault kv get -format=json secret/tsi-rcni/${REGION}/${CLUSTER}/${NS}/${IMGSHA}/${SECNAME}"
+  # vault command:
+  #  vault kv get -format=json secret/tsi-rcni/eu-de/ti-test/trusted-identity/f36b6d491e0a62cb704aea74d65fabf1f7130832e9f32d0771de1d7c727a79cc/dummy
+  # is equivalent of:
+  #  curl -X GET -H "X-Vault-Token: $(vault print token)" ${VAULT_ADDR}/v1/secret/data/tsi-rcni/eu-de/ti-test/trusted-identity/f36b6d491e0a62cb704aea74d65fabf1f7130832e9f32d0771de1d7c727a79cc/dummy
 
   # get the result in JSON format, then convert to string
   # Result can be also an error like this:
   #    No value found at secret/data/tsi-rcni/eu-de/ti-test1/trusted-identity/a8725beade10de172ec0fdbc683/dummyx
-  JRESULT=$($CMD)
+  RESULT_FILE="/tmp/result.$$"
+
+  #JRESULT=$($CMD)
+  SC=$(curl --max-time 10 -s -w "%{http_code}" -H "X-Vault-Request: true" -H "X-Vault-Token: ${VAULT_TOKEN}" -o ${RESULT_FILE}  ${CMD})
   local RT=$?
-  if [ "$RT" == "0" ]; then
+  JRESULT=$(cat $RESULT_FILE)
+  rm -f $RESULT_FILE
+
+  if [[ "$RT" == "28" ]]; then
+    echo "Timeout occured for SECNAME=${SECNAME}"
+    return 1
+  fi
+
```

Comment:
I would like to still check for the passing of the command before going ahead to using the content of it

Reply:
Done